core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -657,10 +657,12 @@ class GroupMetadataManager(brokerId: Int,
       // special handling for version 0:
       // set the expiration time stamp as commit time stamp + server default retention time
       val updatedOffsetAndMetadata =
-        if (offsetAndMetadata.expireTimestamp == org.apache.kafka.common.requests.OffsetCommitRequest.DEFAULT_TIMESTAMP)
-        offsetAndMetadata.copy(expireTimestamp = offsetAndMetadata.commitTimestamp + config.offsetsRetentionMs)
-      else
-        offsetAndMetadata
+        if (offsetAndMetadata.expireTimestamp.contains(OffsetCommitRequest.DEFAULT_TIMESTAMP))
```

Comment:
Related to my previous comment. It would be better if we don't allow both `None` and `Some(-1)`. We should be able to use `None` for both cases.

Reply:
Thanks for the earlier pointer that made me look more closely at this. Though, I'm having a difficult time finding out where we store `-1` for `expireTimestamp` that justifies this check, and having no luck. Any ideas? I'm thinking if a `-1` has been stored (as default) for expiration timestamp, wouldn't we break backward compatibility if we switch this default to `None`?

Reply:
Can you explain why we are doing the `contains` check here if the case is not possible? Should we be using `offset.expireTimestamp.isEmpty` instead?

Reply:
Unless I'm mistaken, I think we can just remove this check and use `offsetAndMetadata`. We shouldn't need the special case for version 0 anymore because it can be treated the same as the latest version. Both use `None` as the expiration timestamp.